File: java/src/main/java/toolgood/algorithm/internals/functions/value/Function_Number.java
package toolgood.algorithm.internals.functions.value;

import java.math.BigDecimal;
import java.util.Map;

import toolgood.algorithm.AlgorithmEngine;
import toolgood.algorithm.Operand;
import toolgood.algorithm.enums.NumberUnitType;
import toolgood.algorithm.enums.OperandType;
import toolgood.algorithm.internals.NumberUnitTypeHelper;
import toolgood.algorithm.internals.functions.Function_0;

public final class Function_Number extends Function_0 {
    private final BigDecimal d;
    private final String unit;

    public Function_Number(BigDecimal func1, String func2) {
        this.d = func1;
        this.unit = func2;
    }

    @Override
    public String Name() {
        return "Num";
    }

    @Override
    public Operand Evaluate(AlgorithmEngine engine, java.util.function.BiFunction<AlgorithmEngine, String, Operand> tempParameter) throws Exception {
        Map<String, NumberUnitType> dict = NumberUnitTypeHelper.GetUnitTypedict();
        BigDecimal d2 = NumberUnitTypeHelper.TransformationUnit(d, dict.get(unit), engine.DistanceUnit, engine.AreaUnit, engine.VolumeUnit, engine.MassUnit);
        return Operand.Create(d2);
    }

    @Override
    public void toString(StringBuilder stringBuilder, boolean addBrackets) {
        stringBuilder.append(d);
        stringBuilder.append(unit);
    }

    @Override
    public OperandType GetResultType() {
        return OperandType.NUMBER;
    }
}

File: JavaScript/src/Internals/Functions/Value/Function_Number.js
import { Function_0 } from '../Function_0.js';
import { Operand } from '../../../Operand.js';
import { NumberUnitTypeHelper } from '../../NumberUnitTypeHelper.js';

class Function_Number extends Function_0 {
    constructor(d, unit) {
        super();
        this.d = d;
        this.unit = unit;
    }

    get Name() {
        return "Num";
    }

    evaluate(engine, tempParameter) {
        let dict = NumberUnitTypeHelper.getUnitTypedict();
        let d2 = NumberUnitTypeHelper.transformationUnit(this.d, dict[this.unit], engine.DistanceUnit, engine.AreaUnit, engine.VolumeUnit, engine.MassUnit);
        return Operand.Create(d2);
    }

    toString2(stringBuilder, addBrackets) {
        stringBuilder.push(this.d);
        stringBuilder.push(this.unit);
    }
}

export { Function_Number };